### DISCUSSION TITLE
Search engine selection in setup flow

### DISCUSSION THREAD

**Discussion by wiki-Bird** [zen-browser/desktop#2213] [Ideas]
In the Zen setup flow, there is a "Choose a search engine" page. This page provides three basic search options Google, DuckDuckGo and Wikipedia. While I understand that these are likely going to cover the vast majority of people, I think it would be nice to provide further options on this page.

<img src="https://github.com/user-attachments/assets/7a9fe1d7-f8ef-4f63-ba83-c5e3a5fd7537" width="400" />

If not wanting to overwhelm the user is a concern, these could be hidden in a dropdown "other" menu, but I do think the choice should be available to people like me who do not use any of the above.

Here are some options that may make sense to add:
- Kagi Search
- Ecosia
- Baidu
- Yandex
- Bing
And, in the same vein as Wikipedia, perhaps YouTube.

Cheers

**Comment by RealDishwash** [comment]
+1 for Brave Search as well

**Comment by mr-cheffy** [comment]
I just didnt have time to add more search engines

**Reply by realSilasYang** [reply]
plz we need youtube and bing

**Reply by wiki-Bird** [reply]
Gotcha. If I have the time I can give it a go myself, I'll open a PR if I do :) thanks for the clarification

**Comment by chikobara** [comment]
we need brave search

**Comment by KodeBarista** [comment]
Surprising to see privacy invading search engines like Google is listed but not other major search engines.

Brave has to be in the list. If not default

**Comment by ajimix** [comment]
Came here also looking for Kagi or for a "Custom" option where I could add Kagi myself or any other one

**Reply by thibaultmol** [reply]
Big +1 
i get that listing every search again isn't possible (although i'll +1 for kagi search) just adding a custom search option would already be a major improvement!

**Comment by dderg** [comment]
You can add Kagi as a search option by following [kagi tutorial for firefox based browsers](https://help.kagi.com/kagi/getting-started/setting-default.html#firefox_browsers)  (will probably work for other search engines too)

**Reply by thibaultmol** [reply]
Oh yeah no, I think most of us know how to do. It's just that we're suggest to have it in the onboarding flow

**Comment by dderg** [comment]
Also noticed that even if you choose other that google search as default during initial setup, it will still keep google as default for private tabs (ironic, isn't it?)

**Reply by wiki-Bird** [reply]
this would make sense if the user chose kagi, fwiw

**Comment by caydog25** [comment]
Cant seem to figure out how to add Bing. I attempted to run the steps for adding kagi but those arent working with latest either.

**Comment by petsud** [comment]
+1 for **Kagi**, great search engine that needs more exposure.

**Reply by antton-dev** [reply]
Kagi isn't free. I think it's kinda weird to promote a paid search engine, especially as Zen is open source

**Reply by thibaultmol** [reply]
A lot of people in this thread have mentioned Kagi (including myself), there's clearly demand for it.
I think just having an asterisk with a mouseover text that says it's a paid option, could be a good solution.

**Reply by petsud** [reply]
Agree with [thibaultmol](https://github.com/thibaultmol). It's their "business model" to make sure they are and will be ad-free.

**Reply by antton-dev** [reply]
I never said the opposite. I'm just saying that adding it on the welcome screen as a **main** search engine is too much because it's a paid engine. And to me, paid world has nothing to do in open source world.
But this is just my opinion, maybe I'm wrong ;)

**Reply by dderg** [reply]
other search engines are also commercial products, they just make money differently
